From 7489e3f6be8d5bd6771dc73b523972a77d00f899 Mon Sep 17 00:00:00 2001
From: Lily Zhang <r58066@freescale.com>
Date: Fri, 30 Apr 2010 15:30:48 +0800
Subject: [PATCH] ENGR00123071 MX53: Remove iomux warning in boot

1. Remove iomux warning in boot
2. Put I2C PIN configurations into table

Signed-off-by: Lily Zhang <r58066@freescale.com>
---
 arch/arm/mach-mx5/mx53_evk_gpio.c |   84 ++++++++++++++-----------------------
 1 files changed, 32 insertions(+), 52 deletions(-)

diff --git a/arch/arm/mach-mx5/mx53_evk_gpio.c b/arch/arm/mach-mx5/mx53_evk_gpio.c
index 49fa5d4..a873518 100644
--- a/arch/arm/mach-mx5/mx53_evk_gpio.c
+++ b/arch/arm/mach-mx5/mx53_evk_gpio.c
@@ -79,9 +79,6 @@ static struct mxc_iomux_pin_cfg __initdata mxc_iomux_pins[] = {
 	 MX53_PIN_EIM_D26, IOMUX_CONFIG_GPIO,
 	 },
 	{
-	 MX53_PIN_EIM_D28, IOMUX_CONFIG_ALT3,
-	 },
-	{
 	 MX53_PIN_EIM_D29, IOMUX_CONFIG_ALT3,
 	 },
 	{
@@ -172,14 +169,8 @@ static struct mxc_iomux_pin_cfg __initdata mxc_iomux_pins[] = {
 	 MX53_PIN_KEY_ROW1, IOMUX_CONFIG_ALT2,
 	 },
 	{
-	 MX53_PIN_KEY_COL3, IOMUX_CONFIG_ALT4,
-	 },
-	{
 	 MX53_PIN_CSI0_D7, IOMUX_CONFIG_ALT5,
 	 },
-	{
-	 MX53_PIN_CSI0_D9, IOMUX_CONFIG_ALT5,
-	 },
 	{ /* UART1 Tx */
 	 MX53_PIN_CSI0_D10, IOMUX_CONFIG_ALT2,
 	 (PAD_CTL_HYS_ENABLE | PAD_CTL_PKE_ENABLE | PAD_CTL_PUE_PULL |
@@ -600,6 +591,38 @@ static struct mxc_iomux_pin_cfg __initdata mxc_iomux_pins[] = {
 	 MX53_PIN_FEC_MDC, IOMUX_CONFIG_ALT0,
 	 PAD_CTL_DRV_HIGH,
 	 },
+	 { /* I2C1 SDA */
+	 MX53_PIN_CSI0_D8, IOMUX_CONFIG_ALT5 | IOMUX_CONFIG_SION,
+	 (PAD_CTL_SRE_FAST | PAD_CTL_ODE_OPENDRAIN_ENABLE |
+	 PAD_CTL_DRV_HIGH | PAD_CTL_100K_PU |
+	 PAD_CTL_HYS_ENABLE),
+	 MUX_IN_I2C1_IPP_SDA_IN_SELECT_INPUT,
+	 INPUT_CTL_PATH0,
+	 },
+	 { /* I2C1 SCL */
+	 MX53_PIN_CSI0_D9, IOMUX_CONFIG_ALT5 | IOMUX_CONFIG_SION,
+	 (PAD_CTL_SRE_FAST | PAD_CTL_ODE_OPENDRAIN_ENABLE |
+	 PAD_CTL_DRV_HIGH | PAD_CTL_100K_PU |
+	 PAD_CTL_HYS_ENABLE),
+	 MUX_IN_I2C1_IPP_SCL_IN_SELECT_INPUT,
+	 INPUT_CTL_PATH0,
+	 },
+	 { /* I2C2 SDA */
+	 MX53_PIN_KEY_ROW3, IOMUX_CONFIG_ALT4 | IOMUX_CONFIG_SION,
+	 (PAD_CTL_SRE_FAST | PAD_CTL_ODE_OPENDRAIN_ENABLE |
+	 PAD_CTL_DRV_HIGH | PAD_CTL_100K_PU |
+	 PAD_CTL_HYS_ENABLE),
+	 MUX_IN_I2C2_IPP_SDA_IN_SELECT_INPUT,
+	 INPUT_CTL_PATH0,
+	 },
+	 { /* I2C1 SCL */
+	 MX53_PIN_KEY_COL3, IOMUX_CONFIG_ALT4 | IOMUX_CONFIG_SION,
+	 (PAD_CTL_SRE_FAST | PAD_CTL_ODE_OPENDRAIN_ENABLE |
+	 PAD_CTL_DRV_HIGH | PAD_CTL_100K_PU |
+	 PAD_CTL_HYS_ENABLE),
+	 MUX_IN_I2C2_IPP_SCL_IN_SELECT_INPUT,
+	 INPUT_CTL_PATH0,
+	 },
 };
 
 static struct mxc_iomux_pin_cfg __initdata mx53_evk_iomux_pins[] = {
@@ -800,49 +823,6 @@ void __init mx53_evk_io_init(void)
 	gpio_request(IOMUX_TO_GPIO(MX53_PIN_GPIO_16), "gpio7_11");
 	gpio_direction_input(IOMUX_TO_GPIO(MX53_PIN_GPIO_16));	/*PMIC_INT*/
 
-
-	/* i2c1 SDA */
-	mxc_request_iomux(MX53_PIN_CSI0_D8,
-			  IOMUX_CONFIG_ALT5 | IOMUX_CONFIG_SION);
-	mxc_iomux_set_input(MUX_IN_I2C1_IPP_SDA_IN_SELECT_INPUT,
-			    INPUT_CTL_PATH0);
-	mxc_iomux_set_pad(MX53_PIN_CSI0_D8, PAD_CTL_SRE_FAST |
-			  PAD_CTL_ODE_OPENDRAIN_ENABLE |
-			  PAD_CTL_DRV_HIGH | PAD_CTL_100K_PU |
-			  PAD_CTL_HYS_ENABLE);
-
-	/* i2c1 SCL */
-	mxc_request_iomux(MX53_PIN_CSI0_D9,
-			  IOMUX_CONFIG_ALT5 | IOMUX_CONFIG_SION);
-	mxc_iomux_set_input(MUX_IN_I2C1_IPP_SCL_IN_SELECT_INPUT,
-			    INPUT_CTL_PATH0);
-	mxc_iomux_set_pad(MX53_PIN_CSI0_D9, PAD_CTL_SRE_FAST |
-			  PAD_CTL_ODE_OPENDRAIN_ENABLE |
-			  PAD_CTL_DRV_HIGH | PAD_CTL_100K_PU |
-			  PAD_CTL_HYS_ENABLE);
-
-	/* i2c2 SDA */
-	mxc_request_iomux(MX53_PIN_KEY_ROW3,
-			  IOMUX_CONFIG_ALT4 | IOMUX_CONFIG_SION);
-	mxc_iomux_set_input(MUX_IN_I2C2_IPP_SDA_IN_SELECT_INPUT,
-			    INPUT_CTL_PATH0);
-	mxc_iomux_set_pad(MX53_PIN_KEY_ROW3,
-			  PAD_CTL_SRE_FAST |
-			  PAD_CTL_ODE_OPENDRAIN_ENABLE |
-			  PAD_CTL_DRV_HIGH | PAD_CTL_100K_PU |
-			  PAD_CTL_HYS_ENABLE);
-
-	/* i2c2 SCL */
-	mxc_request_iomux(MX53_PIN_KEY_COL3,
-			  IOMUX_CONFIG_ALT4 | IOMUX_CONFIG_SION);
-	mxc_iomux_set_input(MUX_IN_I2C2_IPP_SCL_IN_SELECT_INPUT,
-			    INPUT_CTL_PATH0);
-	mxc_iomux_set_pad(MX53_PIN_KEY_COL3,
-			  PAD_CTL_SRE_FAST |
-			  PAD_CTL_ODE_OPENDRAIN_ENABLE |
-			  PAD_CTL_DRV_HIGH | PAD_CTL_100K_PU |
-			  PAD_CTL_HYS_ENABLE);
-
 	/* headphone_det_b */
 	mxc_request_iomux(MX53_PIN_ATA_DATA5, IOMUX_CONFIG_GPIO);
 	mxc_iomux_set_pad(MX53_PIN_ATA_DATA5, PAD_CTL_100K_PU);
-- 
1.5.4.4

